modules/schain.js

```diff
@@ -181,7 +177,16 @@ export function makeBidRequestsHook(fn, bidderRequests) {
     bidderRequest.bids.forEach(bid => {
       let result = resolveSchainConfig(schainConfig, bidder);
       if (result) {
-        bid.schain = deepClone(result);
+        // Initialize ortb2 object if it doesn't exist
+        if (!bid.ortb2) {
+          bid.ortb2 = {};
+        }
+        // Initialize source object if it doesn't exist
+        if (!bid.ortb2.source) {
+          bid.ortb2.source = {};
+        }
+        // Set the schain in ortb2.source.schain
+        bid.ortb2.source.schain = deepClone(result);
       }
     });
```

Comment:
most adapters look for FPD in `bidderRequest.ortb2`. `bid.ortb2` is a copy that would get out of sync with this.

If instead of modifying the bid request objects you hook into `enrichFPD` you can inject FPD for everywhere else (it would then end up in `bid(derRequest).ortb2`, but also RTD modules, auction objects, etc). For example:

https://github.com/prebid/Prebid.js/blob/f8a666af33356a06a8e9a40792b6f432cd4b4790/modules/consentManagementGpp.js#L204-L217